File: src/compiler/parser/visitors/PrimitiveTypedVisitor.java
package compiler.parser.visitors;

import compiler.lexer.token.FloatToken;
import compiler.lexer.token.IntegerToken;
import compiler.lexer.token.StringToken;

public interface PrimitiveTypedVisitor<T> {
  T visit(FloatToken token);

  T visit(IntegerToken token);

  T visit(StringToken token);
}

File: src/compiler/parser/visitors/PrimitiveTypedVisitor.kt
package compiler.parser.visitors

import compiler.lexer.token.FloatToken
import compiler.lexer.token.IntegerToken
import compiler.lexer.token.StringToken

interface PrimitiveTypedVisitor<T> {
    fun visit(token: FloatToken): T
    fun visit(token: IntegerToken): T
    fun visit(token: StringToken): T
}
